Updated Vagrant VMs to Fedora 23

aanm opened the pull request:
Updated Fedora boxes to Fedora 23 on all providers

Signed-off-by: André Martins aanm90@gmail.com

aanm committed:
Updated Vagrant VMs to Fedora 23

Signed-off-by: André Martins <aanm90@gmail.com>

ashcrow commented on cluster/vagrant/provision-master.sh:
```diff
@@ -257,10 +257,10 @@ popd
 # Access it by going to <master-ip>:9090 and login as vagrant/vagrant
 if ! which /usr/libexec/cockpit-ws &>/dev/null; then
-  
+
```
nit: whitespace not required for readability.

ashcrow commented on Vagrantfile:
```diff
@@ -54,16 +54,16 @@ $kube_provider_boxes = {
-      :box_name => 'kube-fedora20',
-      :box_url => 'https://atlas.hashicorp.com/rickard-von-essen/boxes/opscode_fedora-20/versions/0.4.0/providers/parallels.box'
+      :box_name => 'kube-fedora22',
+      :box_url => 'https://opscode-vm-bento.s3.amazonaws.com/vagrant/parallels/opscode_fedora-22_chef-provisionerless.box'
```
Is this endpoint taking over parallels builds from Rickard?

aanm replied:
I saw that Bento has a box for parallels too. Is there a particular reason for using Rickard's instead?

ashcrow replied:
I'm not aware of it. I just noticed his was in use previously is all.

brendandburns replied:
If it doesn't need to change, my preference would be to remain with the other provider, but I don't super care.

brendandburns replied:
actually, nm, since we use the bento build below, this is fine.

aanm replied:
I just saw the comments on lines 54-58 doesn't make sense anymore, should I remove those lines or all (from 49-58)?

ashcrow replied:
I think changing the reference from ```rickard-von-essen/opscode_fedora-20``` to the new name makes the most sense as it's still overwriting the name with ```:box_name```

ashcrow commented:
Shippable build failure seems to be a flake. One build succeeded while the other failed with:

```
Running tests for APIVersion: v1,extensions/v1beta1 with etcdPrefix: kubernetes.io/registry
+++ [1021 04:55:17] Running tests without code coverage
```

aanm commented:
Also, `libvirt` box should be upgrade to fedora-22 or the PR #16000 won't work for `libvirt`

brendandburns commented:
LGTM.  kicked shippable.

ashcrow commented:
Totally unrelated flake for Jenkins e2e.

```
Summarizing 1 Failure:

[Fail] Pods [It] should have their auto-restart back-off timer reset on image update 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/pods.go:910
```

derekwaynecarr commented:
Thanks for the PR!  

I want to test this before LGTM - I think we may need to update how cockpit is installed.

I know there are some users of ```libvirt``` in the community based on past PRs or issues... we may need to check ```dnf``` vs ```yum``` use in the interim.

aanm commented:
@derekwaynecarr Please test it, I've test it with `virtualbox` provider and looks good. `dnf` was introduced in Fedora 20 so I'm assuming that box has it. It would be better to find a new fedora 22 for that provider or #16000 can't be merged (overlayfs appear on kernel 3.18 and fedora 20 is on kernel 3.12). I already asked for box creations for `libvert` (https://github.com/chef/bento/issues/481) is a matter of waiting a little longer or merge this one and, when ready, create a new PR for `libvert` and then merge #16000.

derekwaynecarr commented on cluster/vagrant/provision-master.sh:
```diff
@@ -257,10 +257,10 @@ popd
-  
+
   pushd /etc/yum.repos.d
     wget https://copr.fedoraproject.org/coprs/sgallagh/cockpit-preview/repo/fedora-21/sgallagh-cockpit-preview-fedora-21.repo
```
s/21/22 in above line, but this will need to be branched on fedora version until all providers are updated:

```
wget https://copr.fedoraproject.org/coprs/sgallagh/cockpit-preview/repo/fedora-22/sgallagh-cockpit-preview-fedora-22.repo
```

derekwaynecarr commented:
I just tried on VirtualBox and ran into issues with eth1 not being set properly to 10.245.1.2 - which ultimately made etcd/flannel setup fail in the master.  

I think the issue is back to the following line not being updated:
https://github.com/kubernetes/kubernetes/blob/master/cluster/vagrant/provision-master.sh#L24

Ran out of time to re-test to ensure, but need to check on either 21 or 22 i suspect.

derekwaynecarr commented:
@aanm - are you able to debug the issues in this further?

aanm commented:
@derekwaynecarr Done, I've also updated libvirt too. Since the PR wasn't merged, was causing conflicts and Fedora had reached 23rd I've changed all boxes to fedora 23. I've also found a `libvirt` image from fedora itself, was there any reason to stay with the amadeus one?

derekwaynecarr commented:
@aanm - pulling PR to test locally, thanks

derekwaynecarr commented:
Getting a not found error when downloading: 

`http://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_fedora-23_chef-provisionerless.box`

aanm commented:
@derekwaynecarr Sorry, you're right. I was waiting for a PR to be merged that would allow having that box online. The owner of that PR told me he would need more time to put those boxes online hopefully around Dec 18. So on that time I will ping you.
If you think it's better to merge this PR ASAP I can change it back to F22 and it will work, what do you think?

derekwaynecarr commented:
We can pend until F23 is ready.  Poke when you are ready for another review.

On Thu, Dec 10, 2015 at 12:49 PM, André Martins <notifications@github.com>
wrote:

> @derekwaynecarr <https://github.com/derekwaynecarr> Sorry, you're right.
> I was waiting for a PR to be merged that would allow having that box
> online. The owner of that PR told me he would need more time to put those
> boxes online hopefully around Dec 18. So on that time I will ping you.
> If you think it's better to merge this PR ASAP I can change it back to F22
> and it will work, what do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15998#issuecomment-163700657>
> .
>

derekwaynecarr commented:
I just tested and got the following in VirtualBox, reporting for your awareness, but will need more time to debug than I have today.  If you can test on VirtualBox as well, it would be appreciated.

```
....
==> master: Configuring and enabling network interfaces...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

ARPCHECK=no /sbin/ifup eth1 2> /dev/null

Stdout from the command:

ERROR    : [/etc/sysconfig/network-scripts/ifup-eth] Device eth1 does not seem to be present, delaying initialization.


Stderr from the command:
```

jayunit100 commented:
I see a similar issue (i think) to derek's, `if_to_edit` unbound variable.  This indicates that the ifup stuff is failing.  

I think the root cause is the private_network feature : It doesnt always work on all the vagrant boxes, and possibly its broke on the current f23 box being used here.

aanm commented:
@jayunit100 Thanks for testing it! You test it with VirtualBox, right? I'm still waiting for an issue to be fixed for VMWare provider: https://github.com/chef/bento/issues/521

Once that is fixed I can test it with VirtualBox and libvirt. Otherwise we can test and merge it for VirtualBox and libvirt providers and keep fedora 21 for VMWare for now.

aanm commented:
@jayunit100 @derekparker I've made some changes, test it with VirtualBox and works.

The problem was here:
```
-if [[ "$(grep 'VERSION_ID' /etc/os-release)" =~ ^VERSION_ID=21 ]]; then
-  # Workaround to vagrant inability to guess interface naming sequence
-  # Tell system to abandon the new naming scheme and use eth* instead
-  rm -f /etc/sysconfig/network-scripts/ifcfg-enp0s3
```

Do you want to wait for the VMWare issue (chef/bento#521) to be solved or should I change to a different version of fedora-23?

jayunit100 commented:
thanks for patching, I'll take a look today!

jayunit100 commented:
Works :) . I also confirmed *flannel* starts properly... 
```
Jan 17 18:53:46 kubernetes-master flanneld[3119]: go-etcd2016/01/17 18:53:46 DEBUG: send.request.to http://10.245.1.2:4379/v2/keys/coreos.com/network/subnets?recursive=true&wait=true&waitIndex=7 | method GET
Jan 17 18:53:46 kubernetes-master flanneld[3119]: I0117 18:53:46.225032    3119 udp.go:247] Subnet added: 10.246.101.0/24
Jan 17 18:53:46 kubernetes-master flanneld[3119]: go-etcd2016/01/17 18:53:46 DEBUG: recv.response.from
Jan 17 18:53:46 kubernetes-master flanneld[3119]: go-etcd2016/01/17 18:53:46 DEBUG: recv.success
Jan 17 18:53:46 kubernetes-master flanneld[3119]: go-etcd2016/01/17 18:53:46 DEBUG: recv.response.from
```
Which indicates that indeed the `FLANNEL_ETCD="http://10.245.1.2:4379/-iface=enp0s8"` IP address stuff is working properly.

aanm commented:
Can someone test with parallels? I've test with libvirt but I had to add this `dnf install -y wget` because the libvirt fedora 23 box doesn't have `wget` installed. However I got some errors but I don't know if they are related to `libvirt` fedora box or kube:

```
==> master:   This can take some time based on your network, disk, and cpu speed.
==> master: [WARNING ] Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily
==> master: [ERROR   ] Command 'systemctl status docker.service' failed with return code: 3
==> master: [ERROR   ] output: * docker.service - Docker Application Container Engine
==> master:    Loaded: loaded (/usr/lib/systemd/system/docker.service; disabled; vendor preset: disabled)
==> master:   Drop-In: /usr/lib/systemd/system/docker.service.d
==> master:            `-flannel.conf
==> master:    Active: inactive (dead)
==> master:      Docs: http://docs.docker.com
==> master: [ERROR   ] Command 'systemctl status docker.service' failed with return code: 3
==> master: [ERROR   ] output: * docker.service - Docker Application Container Engine
==> master:    Loaded: loaded (/usr/lib/systemd/system/docker.service; disabled; vendor preset: disabled)
==> master:   Drop-In: /usr/lib/systemd/system/docker.service.d
==> master:            `-flannel.conf
==> master:    Active: inactive (dead)
==> master:      Docs: http://docs.docker.com
==> master: [ERROR   ] Command 'systemctl status kubelet.service' failed with return code: 3
==> master: [ERROR   ] output: * kubelet.service - Kubernetes Kubelet Server
==> master:    Loaded: loaded (/usr/lib/systemd/system/kubelet.service; disabled; vendor preset: disabled)
==> master:    Active: inactive (dead)
==> master:      Docs: https://github.com/GoogleCloudPlatform/kubernetes
==> master: [ERROR   ] Command 'systemctl status kubelet.service' failed with return code: 3
==> master: [ERROR   ] output: * kubelet.service - Kubernetes Kubelet Server
==> master:    Loaded: loaded (/usr/lib/systemd/system/kubelet.service; disabled; vendor preset: disabled)
==> master:    Active: inactive (dead)
==> master:      Docs: https://github.com/GoogleCloudPlatform/kubernetes
==> master: [WARNING ] Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'rading the file unnecessarily
==> master: [WARNING ] Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'rading the file unnecessarily
==> master: [ERROR   ] Command '/opt/kubernetes/helpers/services bounce kube-master-addons' failed with return code: 1
==> master: [ERROR   ] stderr: Failed to execute operation: Connection timed out
==> master: [ERROR   ] retcode: 1
==> master: [ERROR   ] {'pid': 5136, 'retcode': 1, 'stderr': 'Failed to execute operation: Connection timed out', 'std
==> master: [ERROR   ] Command 'systemctl status kube-addons.service' failed with return code: 3
==> master: [ERROR   ] output: * kube-addons.service - Kubernetes Addon Object Manager
==> master:    Loaded: loaded (/usr/lib/systemd/system/kube-addons.service; enabled; vendor preset: disabled)
==> master:    Active: inactive (dead) since Mon 2016-01-18 03:17:27 UTC; 119ms ago
==> master:      Docs: https://github.com/GoogleCloudPlatform/kubernetes
==> master:   Process: 5336 ExecStart=/etc/kubernetes/kube-addons.sh (code=killed, signal=TERM)
==> master:  Main PID: 5336 (code=killed, signal=TERM)
==> master: 
==> master: Jan 18 03:17:23 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:24 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:25 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:25 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:26 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:26 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:27 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:27 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:27 kubernetes-master systemd[1]: Stopping Kubernetes Addon Object Manager...
==> master: Jan 18 03:17:27 kubernetes-master systemd[1]: Stopped Kubernetes Addon Object Manager.
==> master: [ERROR   ] Command 'systemctl status kube-addons.service' failed with return code: 3
==> master: [ERROR   ] output: * kube-addons.service - Kubernetes Addon Object Manager
==> master:    Loaded: loaded (/usr/lib/systemd/system/kube-addons.service; enabled; vendor preset: disabled)
==> master:    Active: inactive (dead) since Mon 2016-01-18 03:17:27 UTC; 193ms ago
==> master:      Docs: https://github.com/GoogleCloudPlatform/kubernetes
==> master:   Process: 5336 ExecStart=/etc/kubernetes/kube-addons.sh (code=killed, signal=TERM)
==> master:  Main PID: 5336 (code=killed, signal=TERM)
==> master: 
==> master: Jan 18 03:17:23 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:24 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:25 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:25 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:26 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:26 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:27 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:27 kubernetes-master kube-addons.sh[5336]: The connection to the server localhost:8080 was re
==> master: Jan 18 03:17:27 kubernetes-master systemd[1]: Stopping Kubernetes Addon Object Manager...
==> master: Jan 18 03:17:27 kubernetes-master systemd[1]: Stopped Kubernetes Addon Object Manager.
^C==> master: Waiting for cleanup before exiting...
^C==> master: Exiting immediately, without cleanup!
```

@jayunit100 
I could remove VMWare from this patch but wouldn't the changes made under `cluster/vagrant/provision-master.sh` break with fedora 21? Unless I write a condition for each particular fedora, is that ok?

jayunit100 commented:
i think:
- rolling out the f23 update for vbox is important,
- older fedoras breaking is ok
- different providers breaking is ok ...

... they can catch up once someone decides to use them.

I realize this is a dangerous opinion... So let's see if there Any other opinions on this?

derekwaynecarr commented:
I will test this, and then assuming I see what @jayunit100 saw, I will merge.  I agree that bumping vbox is a priority and other providers can play catch-up.

aanm commented:
Sounds good, before you merge it, let me know if I should keep the `box_url` for the other providers or if I should not replace the old ones.

derekwaynecarr commented:
I tried this on Fedora using VirtualBox version 4.3 and got the following:

```
==> master: Configuring and enabling network interfaces...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

ARPCHECK=no /sbin/ifup eth1 2> /dev/null

Stdout from the command:

ERROR    : [/etc/sysconfig/network-scripts/ifup-eth] Device eth1 does not seem to be present, delaying initialization.


Stderr from the command:
```

I repulled the box to ensure that was not the issue.

What environment did you guys test on @jayunit100 @aanm ?

derekwaynecarr commented:
I bumped Vagrant to 1.7.4 and got past the error I was seeing.

Please update here to >= 1.7.4
https://github.com/kubernetes/kubernetes/blob/master/Vagrantfile#L8

And update prerequisites section here:
https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/vagrant.md

derekwaynecarr commented:
I want to merge this now as-is.  I can send a follow-up with the documentation changes.

aanm commented:
@derekwaynecarr I applied your suggestions. Thanks!

marun commented:
Oops, I missed this PR in fixing my libvirt problem: #19841  I took a different approach in a couple of places and would like make sure I understand why.

marun commented on cluster/vagrant/provision-master.sh:
```diff
@@ -22,11 +22,7 @@ set -o pipefail
-  # Tell system to abandon the new naming scheme and use eth* instead
-  rm -f /etc/sysconfig/network-scripts/ifcfg-enp0s3
-
+if [[ "$(grep 'VERSION_ID' /etc/os-release)" =~ ^VERSION_ID=23 ]]; then
```
My intention with this conditional was to limit the fixup to Fedora 21.  The fixup appeared unnecessary (and occasionally problematic) for Fedora 22 and 23, at least for libvirt.  Do the opscode boxes for 23 have different configuration from the fedora libvirt image that does not require this block?

marun commented on cluster/vagrant/provision-master.sh:
```diff
@@ -108,8 +104,8 @@ fi
   pushd /etc/yum.repos.d
-    wget https://copr.fedoraproject.org/coprs/sgallagh/cockpit-preview/repo/fedora-21/sgallagh-cockpit-preview-fedora-21.repo
-    yum install -y cockpit cockpit-kubernetes
+    wget https://copr.fedoraproject.org/coprs/sgallagh/cockpit-preview/repo/fedora-22/sgallagh-cockpit-preview-fedora-22.repo
```
I had a problem with wget being missing on the fedora 23 libvirt image.  Do the opscode boxes include it by default?

aanm replied:
@marun Forgot to reply sorry. No, wget doesn't exit under fedora. See above^

>I've test with libvirt but I had to add this dnf install -y wget because the libvirt fedora 23 box doesn't >have wget installed. However I got some errors but I don't know if they are related to libvirt fedora >box or kube:

marun commented on cluster/vagrant/provision-node.sh:
```diff
@@ -22,11 +22,7 @@ set -o pipefail
-  # Tell system to abandon the new naming scheme and use eth* instead
-  rm -f /etc/sysconfig/network-scripts/ifcfg-enp0s3
-
+if [[ "$(grep 'VERSION_ID' /etc/os-release)" =~ ^VERSION_ID=23 ]]; then
```
Is the fixup this block encloses still required for the opscode boxes?

marun commented:
The last thing I found in #19841 was the need to remove the bogus fedora 23 repo the saltstack boostrap was installing.  Did anyone else notice the warning message on 'dnf update' regarding the missing saltstack repo for fedora 23?

Roconda commented:
Its broken as of today, see:

==> minion-1: file: file:///etc/yum.repos.d/saltstack-salt-fedora-21.repo, line: 1
==> minion-1: '<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n'
==> minion-1:  * ERROR: Failed to run install_fedora_deps()!!!

marun commented:
@Roconda Not sure what virt solution you're using, but #19841 fixes some issues that I had with libvirt.  From the error message, though, it looks like you may be trying to install on Fedora 21.  The salt bootstrap keys to the os version to decide which repo to configure, and I would expect to see 'saltstack-salt-fedora-23.repo' if the os was Fedora 23.

luebken commented:
I'm seeing the same problem as @Roconda. I'm with Vagrant in a Mac.

Just downloaded kubernetes release 1.1.7 and followed [getting started instructions](http://kubernetes.io/v1.1/docs/getting-started-guides/vagrant.html). So from what I can tell this brakes the Getting Started with Vagrant.

aanm commented:
@jcf https://github.com/kubernetes/kubernetes/pull/15998#issuecomment-172295274
